core/src/storage2/lazy/lazy_cell.rs

```diff
@@ -307,6 +308,33 @@ where
     pub fn get_mut(&mut self) -> Option<&mut T> {
         self.load_entry_mut().value_mut().into()
     }
+
+    /// Sets the value in this cell to `value`, without executing any reads.
+    ///
+    /// # Note
+    ///
+    /// No loads will be executed.
+    ///
+    /// # Panics
+    ///
+    /// If accessing the inner value fails.
+    pub fn set(&mut self, new_value: T) {
+        // SAFETY: This is critical because we mutably access the entry.
+        let cache = unsafe { &mut *self.cache.get_ptr().as_ptr() };
+        if cache.is_none() {
+            // Write value to storage
+            let key = self
+                .key
+                .expect("key must exist when attempting to set a value");
```

Comment:
I'm not entirely sure if this assumption always holds true. Probably requires further discussion.